source/http_stream.c

```diff
@@ -138,11 +140,23 @@ static void s_on_body_call(napi_env env, napi_value on_body, void *context, void
         napi_value params[1];
         const size_t num_params = AWS_ARRAY_SIZE(params);
 
+#ifndef NODE_API_NO_EXTERNAL_BUFFERS_ALLOWED
```

Comment:
We could do a runtime check instead. The [Electron docs](https://www.electronjs.org/blog/v8-memory-cage#i-want-to-refactor-a-node-native-module-to-support-electron-21-how-do-i-do-that) say that `napi_create_external_buffer()` will fail. We could just try making the buffer the OTHER way if the first try fails.

The advantage is: the same compiled code will work in Electron/Non-Electron

Just an idea...

Reply:
lol Bret just made the same comment on slack. I will update it to do a runtime check instead.